src/logger.jl

```diff
@@ -0,0 +1,34 @@
+import Logging, Dates
+export ModelLogger, shouldlog, min_enabled_level, catch_exceptions, handle_message
+
+struct ModelLogger <: Logging.AbstractLogger
+    stream::IO
+    min_level::Logging.LogLevel
+    message_limits::Dict{Any,Int}
+end
+ModelLogger(stream::IO=stderr, level=Logging.Info) = ModelLogger(stream, level, Dict{Any,Int}())
+
+Logging.shouldlog(logger::ModelLogger, level, _module, group, id) = get(logger.message_limits, id, 1) > 0
+
+Logging.min_enabled_level(logger::ModelLogger) = logger.min_level
+
+Logging.catch_exceptions(logger::ModelLogger) = false
+
+function Logging.handle_message(logger::ModelLogger, level, message, _module, group, id, filepath, line; maxlog = nothing, kwargs...)
+    if maxlog !== nothing && maxlog isa Integer
+        remaining = get!(logger.message_limits, id, maxlog)
+        logger.message_limits[id] = remaining - 1
+        remaining > 0 || return
+    end
+    buf = IOBuffer()
+    iob = IOContext(buf, logger.stream)
+    level = level == Logging.Warn ? "Warning" : string(level)
+    module_name = something(_module, "nothing")
+    file_name = something(filepath, "nothing")
+    line_number = something(line, "nothing")
+    msg_timestamp = Dates.format(Dates.now(), "[dd/mm/yyyy HH:MM:SS]")
```

Comment:
This is how the system is meant to be used :-)

As you have found out, you can get timestamps for every message in your application by customizing the logging backend rather than messing with the `@info` statements themselves.